gobblin-core/src/main/java/gobblin/qualitychecker/row/RowLevelPolicyChecker.java

```diff
@@ -16,26 +16,33 @@
 import java.io.IOException;
 import java.util.List;
 
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 
+import com.google.common.base.Strings;
 import com.google.common.io.Closer;
 
 import gobblin.configuration.State;
 import gobblin.util.FinalState;
+import gobblin.util.WriterUtils;
 
 
 public class RowLevelPolicyChecker implements Closeable, FinalState {
 
   private final List<RowLevelPolicy> list;
+  private final State state;
+  private final FileSystem fs;
   private boolean errFileOpen;
   private final Closer closer;
   private RowLevelErrFileWriter writer;
 
-  public RowLevelPolicyChecker(List<RowLevelPolicy> list) {
+  public RowLevelPolicyChecker(List<RowLevelPolicy> list, State state) throws IOException {
     this.list = list;
+    this.state = state;
```

Comment:
Can you just save the Id instead of the entire state? id seems to get the only thing used.